core/src/main/java/org/apache/carbondata/core/datamap/dev/IndexFactory.java

```diff
@@ -187,7 +187,8 @@ public void deleteSegmentIndexData(String segmentNo) throws IOException {
    * 4. INDEX_COLUMNS should be exists in table columns
    */
   public void validate() throws MalformedIndexCommandException {
-    List<CarbonColumn> indexColumns = carbonTable.getIndexedColumns(dataMapSchema);
+    List<CarbonColumn> indexColumns =
+        carbonTable.getIndexedColumns(dataMapSchema.getIndexColumns());
```

Comment:
datamap schema should have already validated columns. If columns are dropped, we can update the datamap schema. Everytime checking columns in datamap schema is valid or not from all the flows is not efficient

Reply:
handled